File: WebServicesCorreios/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace WebServicesCorreios
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void btBuscar_Click(object sender, EventArgs e)
        {
            using (var ws = new WSCorreios.AtendeClienteClient())
            {
                try
                {
                    var resultado = ws.consultaCEP(tbCEP.Text);
                    tbEndereco.Text = resultado.end;
                    tbComplemento.Text = resultado.complemento;
                    tbComplemento2.Text = resultado.complemento2;
                    tbCidade.Text = resultado.cidade;
                    tbBairro.Text = resultado.bairro;
                    tbEstado.Text = resultado.uf;
                }
                catch (Exception ex)
                {
                    MessageBox.Show(ex.Message, "Erro", MessageBoxButtons.OK, MessageBoxIcon.Error);
                }
            }
        }
    }
}

File: WebServicesCorreios.VB/Form1.vb
Public Class Form1
    Private Sub BtBuscar_Click(sender As Object, e As EventArgs) Handles BtBuscar.Click
        Using WS = New WSCorreios.AtendeClienteClient()
            Try
                Dim Resultado = WS.consultaCEP(TbCEP.Text)
                TbEndereco.Text = Resultado.[end]
                TbComplemento.Text = Resultado.complemento
                TbComplemento2.Text = Resultado.complemento2
                TbCidade.Text = Resultado.cidade
                TbBairro.Text = Resultado.bairro
                TbEstado.Text = Resultado.uf
            Catch Ex As Exception
                MessageBox.Show(Ex.Message, "Erro", MessageBoxButtons.OK, MessageBoxIcon.[Error])
            End Try
        End Using
    End Sub
End Class
